aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -383,13 +388,13 @@ private String getIdentityValue(String identityClaim) throws ApiException {
         return identityValue;
     }
 
-    private ArrayList<String> getUserGroups() throws ApiException {
-        CognitoUserPoolsAuthProvider cognitoProvider = getAuthProvider();
+    private ArrayList<String> getUserGroups(AuthorizationType authType) throws ApiException {
+        // TODO: this doesn't work with OIDC right now. Implement custom groups claim.
         ArrayList<String> groups = new ArrayList<>();
         final String GROUPS_KEY = "cognito:groups";
 
         try {
-            JSONObject accessToken = CognitoJWTParser.getPayload(cognitoProvider.getLatestAuthToken());
+            JSONObject accessToken = CognitoJWTParser.getPayload(getAuthToken(authType));
```

Comment:
`getPayload` can throw an `CognitoParameterInvalidException`.  Can we catch it and wrap it in an `APIException`?  You could throw the same one that is thrown for a `JsonException` below.